lib/net.js

```diff
@@ -1033,6 +1033,8 @@ function internalConnect(
   self, address, port, addressType, localAddress, localPort, flags) {
   // TODO return promise from Socket.prototype.connect which
   // wraps _connectReq.
+  debug('resume');
+  self.resume();
```

Comment:
We should only resume if flowing? Or we might lose data when reconnecting a paused stream. @mcollina

Reply:
But flowing is always false after connected

Reply:
The problem here is that if the socket is paused or never started, you will force it to start. Causing lost data for the user.

Reply:
I agree with @ronag.
Could you just check if any `data` event is there (which implies flowing)?

Also please consider to add whatever solution you find here in `internalConnectMultiple`, otherwise it will fail with network family autoselection.